cloud-controller-manager/do/kubernetes_lb_validator_test.go

```diff
@@ -0,0 +1,401 @@
+	panic("not supported")
+}
+
+func (f *fakeKubernetesService) AddRegistry(ctx context.Context, req *godo.KubernetesClusterRegistryRequest) (*godo.Response, error) {
+	panic("not supported")
+}
+
+func (f *fakeKubernetesService) RemoveRegistry(ctx context.Context, req *godo.KubernetesClusterRegistryRequest) (*godo.Response, error) {
+	panic("not supported")
+}
+
+func (f *fakeKubernetesService) RunClusterlint(ctx context.Context, clusterID string, req *godo.KubernetesRunClusterlintRequest) (string, *godo.Response, error) {
+	panic("not supported")
+}
+
+func (f *fakeKubernetesService) GetClusterlintResults(ctx context.Context, clusterID string, req *godo.KubernetesGetClusterlintRequest) ([]*godo.ClusterlintDiagnostic, *godo.Response, error) {
+	panic("not supported")
+}
+
+func (f *fakeKubernetesService) ListNodePools(ctx context.Context, clusterID string, opts *godo.ListOptions) ([]*godo.KubernetesNodePool, *godo.Response, error) {
+	cluster := fakeCluster()
+	return cluster.NodePools, newFakeOKResponse(), nil
+}
+
+type fakeRegionsService struct {
+	listFn func(context.Context, *godo.ListOptions) ([]godo.Region, *godo.Response, error)
+}
+
+func (f *fakeRegionsService) List(ctx context.Context, listOpts *godo.ListOptions) ([]godo.Region, *godo.Response, error) {
+	return f.listFn(ctx, listOpts)
+}
+
+func Test_Handle(t *testing.T) {
+	//cluster := fakeCluster()
+	testcases := []struct {
+		name            string
+		req             admission.Request
+		oldObject       runtime.RawExtension
+		gCLient         *godo.Client
+		lbRequest       *godo.LoadBalancerRequest
+		expectedAllowed bool
+		expectedReason  string
+		resp            *godo.Response
+	}{
+		{
+			name: "Allow if service type is not load balancer",
+			req: admission.Request{AdmissionRequest: fakeAdmissionRequest(
+				&corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: "test",
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeNodePort,
+					},
+				},
+				nil,
+			)},
+			expectedAllowed: true,
+		},
+		{
+			name: "Allow if request is of type DELETE",
+			req: admission.Request{AdmissionRequest: fakeAdmissionRequest(
+				&corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: "test",
+						DeletionTimestamp: &metav1.Time{
+							Time: time.Now().UTC(),
+						},
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeLoadBalancer,
+					},
+				},
+				nil,
+			)},
+
+			lbRequest:       fakeValidLBRequest(),
+			expectedAllowed: true,
+		},
+		{
+			name: "Allow CREATE happy path",
+			req: admission.Request{AdmissionRequest: fakeAdmissionRequest(
+				&corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: "test",
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeLoadBalancer,
+					},
+				},
+				nil,
+			)},
+			expectedAllowed: true,
+		},
+		{
+			name: "Deny CREATE happy path",
```

Comment:
This sub-test looks 100% identical to the previous one except for `expectedAllowed` being different, and yet the test does not fail. If I am correct, then this indicates a bug in the test implementation (which I guess is around not comparing all combinations of the expected and actual allowed values as I indicated below).

Same thing for the `Allow Update happy path` and `Deny Update happy path` tests I believe.